### PR TITLE
Update `load_from_checkpoint` docstrings

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Have seen a lot of users using it incorrectly. It's not the same as PyTorch `load_state_dict`. Maybe the examples were not presenting the correct image that it returns the loaded model and is not inplace.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7 @awaelchli @ananthsub @ninginthecloud

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

File: pytorch_lightning/core/saving.py [hunk 1]
```diff
@@ -95,28 +95,32 @@ def load_from_checkpoint(
                 hyperparameter values.
 
         Return:
-            :class:`LightningModule` with loaded weights and hyperparameters (if available).
+            :class:`LightningModule` instance with loaded weights and hyperparameters (if available).
+
+        Note:
+            ``load_from_checkpoint`` is a class method. You should use your :class:`LightningModule`
+            class to call it instead of :class:`LightningModule` instance.
```
Comment (awaelchli) [inline pytorch_lightning/core/saving.py:102]:
```suggestion
            class to call it instead of the :class:`LightningModule` instance.
```

Comment (awaelchli) [reply pytorch_lightning/core/saving.py:102]: We could mark the word "class" in bold :)

**Review by awaelchli — APPROVED** [review]
